### PR TITLE
Fixes #1105 - adds static Log.None property with appropriate tests.  …

### LINKED ISSUE #1105: Expose `SilentLogger` as `Logger.None`
**Does this issue relate to a new *feature* or an existing *bug*?**
- [ ] Bug
- [x ] New Feature

**What version of Serilog is affected?  Please list the related NuGet package.**
2.6.0

**What is the target framework and operating system? See [target frameworks]
- [x ] netCore 2.0
- [ ] netCore 1.0
- [ ] 4.7
- [ ] 4.6.x
- [ ] 4.5.x

**Please describe the current behavior?**
There is no mechanism for creating a Serilog.ILogger that does nothing.

**Please describe the expected behavior?**
I sometimes use an optional pattern for logging in libraries such as:

````csharp
        public SomeClass(ILogger logger = null)
        {
            m_logger = (logger ?? new SilentLogger).ForContext<SomeClass>();
        }
````

This allows consumers of the class that aren't using logging to use the abstraction without logging, while those that are can pass in a logger without having all log statements require a condition method operator.  Being able to instantiate a SilentLogger makes this cheap.  

Is there a better pattern with Serilog for accomplishing this?

### PR DESCRIPTION
**Pull request by IanYates**
**What issue does this PR address?**
Adds a static `Log.None` property as per the discussion in #1105 

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [X] The commit follows our [guidelines](https://github.com/serilog/serilog/blob/dev/CONTRIBUTING.md)
- [X] Unit Tests for the changes have been added (for bug fixes / features)

**Other information**:
Basic change to stash a `SilentLogger` instance in a private static variable on the `Log` class. That instance is used as the initial value of the `Logger` property.  The same instance is then exposed via the `None` property. `CloseAndFlush()` was modified to reset back to this instance rather than creating yet another `SilentLogger` instance.

Comments were updated where appropriate and code style was preserved.

I've added tests to ensure that the `None` property is initialised properly.  I've also ensured via a test that initialisation of the `Logger` property doesn't affect the `None` property. 
Finally I've ensured that `CloseAndFlush` doesn't corrupt the logger stored and accessible via the `None` property by checking that it remains a `SilentLogger` instance and that we can invoke it without raising an exception.

I think there are lines of words here than lines of code changed in the PR.  All tests (on my machine) seem to pass in the various framework versions.

Thanks 💯

### CODE REVIEW

**Comment by IanYates** [comment]
Hmmm - AppVeyor failed in an area that I don't *think* I affected.  `ContextPropertiesCrossAsyncCalls`...  Travis CI seems happy (as did my PC).

**Comment by nblumhardt** [comment]
Thanks, Ian!

Since this is more of a plumbing feature, than something that will be used regularly by callers of `Log`, I'd thought perhaps `Logger.None` rather than `Log.None` might keep it tucked away?

File: src/Serilog/Log.cs [hunk 1]
```diff
@@ -1201,5 +1209,16 @@ public static bool BindProperty(string propertyName, object value, bool destruct
         {
             return Logger.BindProperty(propertyName, value, destructureObjects, out property);
         }
+
+        /// <summary>
+        /// An <see cref="ILogger"/> instance that silently ignores all log messages
```
Comment (nblumhardt) [inline src/Serilog/Log.cs]: Nitpick, could use a `.` here.

Perhaps we could try `"that efficiently ignores all method calls"`?

Comment (IanYates) [reply src/Serilog/Log.cs]: 👍 
Shall do

**Comment by IanYates** [comment]
Not sure why I ended up on Log.None...  Logger.None makes more sense.  Shall amend this evening. Thanks

**Comment by IanYates** [comment]
Closing in favour of PR #1110 instead
